cookbooks/bcpc-hadoop/attributes/yarn.rb

```diff
@@ -1,3 +1,46 @@
+require 'shellwords'
+
+default[:bcpc][:hadoop][:yarn][:env_sh].tap do |env_sh|
+  env_sh[:YARN_LOG_DIR] = '/var/log/hadoop-yarn'
+  env_sh[:YARN_PID_DIR] = '/var/run/hadoop-yarn'
+  env_sh[:YARN_IDENT_STRING] = 'yarn'
+  env_sh[:HADOOP_YARN_USER] = 'yarn'
+  env_sh[:YARN_CONF_DIR] = '$YARN_HOME/conf'
+  env_sh[:JAVA_HOME] = node[:bcpc][:hadoop][:java]
+  env_sh[:JAVA] = File.join(env_sh[:JAVA_HOME], 'bin', 'java')
+  env_sh[:YARN_HEAPSIZE] = '-Xmx1000m'
+  env_sh[:JAVA_HEAP_MAX] = env_sh[:YARN_HEAPSIZE]
```

Comment:
There is an if conditional here
https://github.com/bloomberg/chef-bach/blob/master/cookbooks/bcpc-hadoop/templates/default/hdp_yarn-env.sh.erb#L32 otherwise we get Invalid maximum heap size: -Xmx-Xmx1000mm